### PR TITLE
Fix: avoid matching capture pattern name ( Fixes #291)

### LINKED ISSUE #291: Using the exact name and syntax of a URL capture causes a runtime error
Minimal reproduction:

```haskell
module Main (main) where

import Web.Scotty

main :: IO ()
main = scotty 3000 $
  get "/:some_named_capture" $ do
    val <- param "some_named_capture"
    text val
```

When requesting the url "/:some_named_capture" exactly, the value is not captured at all as it matches literally. This results in the `param` call failing.

I actually discovered this whilst reading through scotty's source and noticing the order of the guards [in the go function within matchRoute](https://github.com/scotty-web/scotty/blob/ff4fa83f805923c3be902bfb896ddecd8259d016/Web/Scotty/Route.hs#L124). Specifically, the `equal literals` case is checked before the `p is a capture` case, which should probably be first for the expected behaviour.

### PR DESCRIPTION
**Pull request by jfraudeau**
As stated in the issue #291, GET /test/:capture would match a capture defined as `get "/test/:capture" _handler` with the `capture` absent from params.

With this fix the param would be set to literally `":capture`

### CODE REVIEW

**Comment by ocramz** [comment]
Thank you @jfraudeau ! Could you please add a test for this as well?

**Comment by jfraudeau** [comment]
Test added @ocramz thanks for your time

File: test/Web/ScottySpec.hs [hunk 1]
```diff
@@ -56,6 +56,9 @@ spec = do
           it ("properly handles extra slash routes for " ++ method ++ " requests") $ do
             makeRequest "//scotty" `shouldRespondWith` 200
 
+        withApp (route "/:paramName" $ param "paramName" >>= text) $ do
```
Comment (ocramz) [inline test/Web/ScottySpec.hs:59]: Route parameters are now called `captureParam`

File: test/Web/ScottySpec.hs [hunk 2]
```diff
@@ -56,6 +56,9 @@ spec = do
...
+          it ("captures route parameters for " ++ method ++ " requests when parameter matches its name") $ do
+            makeRequest "/:paramName" `shouldRespondWith` ":paramName"
```
Comment (ocramz) [inline test/Web/ScottySpec.hs:61]: I don't think this is the correct behaviour.  

```haskell
makeRequest "/:x" `shouldRespondWith` "x" 
```
  instead, without the colon.

**Review by ocramz — CHANGES_REQUESTED** [review]
Thank you, but I don't think the implemented behaviour is correct, see comments on the test.

File: test/Web/ScottySpec.hs (hunk 2 above)
Comment (jfraudeau) [reply test/Web/ScottySpec.hs:61]: Hi sorry for the delayed response, I didn't notice your comments until now.
I believe the current test is correct as the colon is part of the value. From the point of view of the user the colon has no special meaning.

File: test/Web/ScottySpec.hs (hunk 1 above)
Comment (jfraudeau) [reply test/Web/ScottySpec.hs]: fixes

**Review by ocramz — COMMENTED** [review]
Sorry for taking so long @jfraudeau , would you mind adding a changelog entry for this as well? Thank you!

**Comment by ocramz** [comment]
Oh and please fix the conflict in the test files. Thanks again!

**Comment by jfraudeau** [comment]
All done, I hope the changelog entry is clear enough

**Review by ocramz — APPROVED** [review]
All good thank you @jfraudeau  ! 👍

**Pull request merged**